backend/jottacloud/jottacloud.go

```diff
@@ -286,12 +290,27 @@ machines.`)
 		if err != nil {
 			return nil, err
 		}
-		return fs.ConfigChooseExclusive("choose_device_result", "config_device", `Please select the device to use. Normally this will be Jotta`, len(acc.Devices), func(i int) (string, string) {
-			return acc.Devices[i].Name, ""
+
+		deviceNames := make([]string, len(acc.Devices))
+		for i, dev := range acc.Devices {
+			deviceNames[i] = dev.Name
+			if i > 0 && deviceNames[i] == defaultDevice {
+				// Move the special Jotta device to the top, which will also make it the default choice.
```

Comment:
A bubble sort style move!

Check out https://github.com/golang/go/wiki/SliceTricks#insert for more efficient ways of doing this, something like

```go
copy(deviceNames[1:i], deviceNames[0:i-1])
deviceNames[0] = dev.Name
```

Reply:
Aha. Needs some free time to be able to run through a manual test again, but will look into it.

Reply:
Don't feel you have to change it. It is very clear what the code is doing which is the important thing.